src/extension/serializer.ts

```diff
@@ -355,6 +355,11 @@ export abstract class SerializerBase implements NotebookSerializer, Disposable {
     }
 
     data.cells.forEach((cell) => {
+      if (identity === RunmeIdentity.UNSPECIFIED && cell.metadata?.['runme.dev/originalId']) {
```

Comment:
We should try to make `reconcileCellIdentity` go away instead of further expanding the quick fix band-aid that it was.